lib/perl/Genome/Model/Build/Command/SymlinkResults.pm

```diff
@@ -0,0 +1,50 @@
+package Genome::Model::Build::Command::SymlinkResults;
+
+use strict;
+use warnings;
+
+use Genome;
+
+class Genome::Model::Build::Command::SymlinkResults {
+    is => ['Command::V2'],
+    has => [
+        build => {
+            is => 'Genome::Model::Build',
+            shell_args_position => 1,
+            doc => "The builds that has results you'd like to create symlinks to",
+        },
+        destination => {
+            is => 'DirectoryPath',
+            shell_args_position => 2,
+            doc => 'The desired location under which to make a directory of symlink(s)',
+        }
+    ],
+    doc => 'Creates symlinks to the results of a build.',
+};
+
+sub help_detail {
+    my $self = shift;
+    return <<EOP;
+Creates symlinks to the results of a build.  The <destination> option should
+be an existing directory.
+EOP
+}
+
+sub execute {
+    my $self = shift;
+    my $build = $self->build;
+    my $destination = $self->destination;
+
+    Genome::Sys->validate_directory_for_read_write_access($destination);
+
+    unless($build->can('symlink_results')) {
+        $self->fatal_message('Symlinking results is unavailable for builds of type "%s"', $build->type_name);
+    }
+
+    my $dir = $build->symlink_results($destination);
+    $self->status_message('Results for build %s symlinked to %s', $build->__display_name__, $dir);
+
```

Comment:
Should most of this code be part of symlink_results instead of special knowledge on the command level?  The same comment goes for the validate_directory_for_read_write_access() call; it looks like knowledge about how to do symlinking exists in two different abstraction layers.  Does it make sense to validate the directory before figuring out if this is a build type that even supports symlinking?

Since each build will have to do its own flavor of symlinking, an alternative pattern as per recommendations in POODR would be to have the build classes implement a handler, an interface dictated by the base class that would default to the fatal message I mentioned above:

```perl
sub symlink_results {
    my $self          = shift;
    my $destination   = shift;

    my $dir = $self->perform_symlinking($destination);  # the callback
    $self->status_message('Results for build %s symlinked to %s', $self->__display_name__, $dir);
}
```
Another note... Given the interface here, is it accurate to say that any error that occurs in this process is a thrown exception?  For the interface portion of the command, would it be useful to catch the exception and convert to a fatal_message?